### PR TITLE
Document the use of MPI_COMM_WORLD in MPI_InitFinalize.

### PR DESCRIPTION
**Pull request by bangerth**
This resolves one of the two remaining issues identified by @kronbichler in #6016.

### CODE REVIEW

File: include/deal.II/base/mpi.h [hunk 1]
```diff
@@ -357,24 +357,45 @@ namespace Utilities
...
+     * compiler will invoke the destructor of this object which in
+     * turns calls <code>MPI_Finalize</code> to shut down the MPI
+     * system.
+     *
+     * This class is used in step-17, step-18, step-40, step-32, and
+     * several others.
+     *
+     * @note This class performs initialization of the MPI subsystem
+     * as well as the dependent libraries listed above through the
+     * `MPI_COMM_WORLD` communicator. This means that you will have to
```
Comment (kronbichler) [inline include/deal.II/base/mpi.h]: Do our doxygen scripts understand `? 

I would write `<code>MPI_COMM_WORLD</code>` to be consistent with the rest of the file.

Comment (bangerth) [reply include/deal.II/base/mpi.h]: Yes, and we've started using that in a number of places already.

**Comment by bangerth** [comment]
I already count 130 lines in the `include/` directory where we use backticks. Do you really want me to change this patch, or has that train left the station already?

**Comment by kronbichler** [comment]
You're right, the rain has left that station. And anyone on github will understand those ticks, and people with intimate github knowledge are just becoming more numerous.

**Comment by bangerth** [comment]
It's not actually github -- it's just markdown formatting. Github just happens to use markdown, but so do many other places on the internet.

**Pull request merged**